docs/scarpet/api/Inventories.md

```diff
@@ -206,3 +206,138 @@ Returns size of the actual dropped items.
+`action` is a string corresponding to the interaction type.
+Can be one of:
+
+Slot interactions:
+* `pickup`
+* `quick_move`
+* `swap`
+* `clone`
+* `throw`
+* `quick_craft`
+* `pickup_all`
+
+Other interactions:
+* `button` Pressing a button in certain screens that have then (enchantment table, lectern, loom and stonecutter)
+* `close` Triggers when the screen gets closed.
+
+`index` is the slot index of the slot interaction.
+If clicked outside the screen (where it would drop held items), this value is -999.
+In the case of a button interaction, this is the button index.
+Note that for lecterns, this index can be certain a value above 100, for jumping to a certain page.
+This can come from formatted text inside the book, with a `change_page` click event action.
+
+`button` corresponds to the mouse button used for the slot click event.
+This only works for slot click interactions, for `button` and `close` type, this value is 0.
+
+0 -> left click
+1 -> right click
+2 -> middle click
+3, 4 -> extra mouse buttons (don't work reliably, only if holding an item in the cursor slot)
+
+By returning a string `'cancel'` in the callback function,
+the screen interaction can be cancelled.
+This doesn't work for the `close` action.
+
+The `create_screen` function returns a `screen_handler` value,
+which can be used in `inventory_size()`, `inventory_has_items()`,
+`inventory_get()` and `inventory_set()` to access the screens slots.
+In addition to that, when using `inventory_get()` or `inventory_set()`,
+when using `-1` as the slot, the cursor stack will be accessed. 
+
+Example usage:
+```py
+global_screen = create_screen(player,'generic_9x6',format('db Test'),_(screen, player, action, index, button) -> (
+    print(player('all'),str('%s\n%s\n%d\n%d',player,action,index,button)); //for testing
+    if(action=='pickup',
+        inventory_set(screen,index,1,if(inventory_get(screen,index),'air','red_stained_glass_pane'));
+    );
+    'cancel'
+));
+```
+
+
+### `close_screen(screen)`
+
+Closes the screen of the given screen handler value.
+Returns `true` if the screen was closed.
+If the screen is already closed, returns `false`.
+
+### `screen_property(screen, property)`
+
+### `screen_property(screen, property, value)`
+
+Queries or modifies a certain `property` of a `screen`.
+The `property` is a string with the name of the property.
+When called with `screen` and `property` parameter, returns the current value of the property.
+When specifying a `value`,
+the property will be assigned the new `value` and synced with the client.
+
+**Options for `property` string:**
+
+| `property` | Required screen type | Type | Description |
+|---|---|---|---|
+| `name` | **All** | text | The name of the screen, as specified in  `create_screen()` function. Can only be queried. |
+| `open` | **All** | boolean | Returns `true` if the screen is open, `false` otherwise. Can only be queried. |
+| `fuel_progress` | furnace | number | Current value of the fuel indicator. |
```

Comment:
its only furnace, or applies to other cooking aparati, like blast furnace, etc?

Reply:
ok, blast furnace and smokers are not supported, right?

Reply:
Since they are exactly the same screen, with only server side behaviour differences, i think it makes sense to only implement the `AbstractFurnaceScreenHandler`.